### PR TITLE
fix: extensions have an leb128 opcode

### LINKED ISSUE #76: LEB code for extensions operators
See WebAssembly/spec#1228 (comment): prefix is a single byte, and what follows is a LEB128

The changes in the https://github.com/WebAssembly/spec/pull/1235/files#diff-59c366fbc32de0cb02b330bf7871317f shows that we cannot just read a byte -- operator can be represented as extended LEB

### PR DESCRIPTION
**Pull request by ngzhian**
Fixed #76

### CODE REVIEW

File: src/WasmParser.ts [hunk 1]
```diff
@@ -2073,7 +2073,12 @@ export class BinaryReader {
   }
 
   private readCodeOperator_0xfc(): boolean {
-    var code = this._data[this._pos++] | 0xfc00;
+    var pos = this._pos;
+    // u32 must not exceed ceil(N/7) = 5 bytes, + 1 prefix byte.
+    if (!this._eof && pos + 1 + 5 > this._length) {
```
Comment (yurydelendik) [inline src/WasmParser.ts:2078]: `this.hasVarIntBytes()` needs to be used here instead (no `this._eof`-padding check is needed)

Comment (yurydelendik) [reply src/WasmParser.ts:2078]: also _xfd and _xfe missing this check too.

Comment (ngzhian) [reply src/WasmParser.ts:2078]: 0xfd and 0xfe have a check for MAX_CODE_OPERATOR_?_SIZE, do they still need to be around if we also have with `hasVarIntBytes`? (With leb128 opcodes, those MAX_CODE_OPERATOR_?_SIZE needs to be updated too.)

Comment (yurydelendik) [reply src/WasmParser.ts:2078]: >  those MAX_CODE_OPERATOR?_SIZE needs to be updated too.

Not really, these are "does it need more bytes check" for chunked parsing. So it rather checks if more data is need and unrelated to the subject.

> `hasVarIntBytes`

Agree that we need separate less generic hasVar32Bytes for check.

Comment (yurydelendik) [reply src/WasmParser.ts]: Okay, I pushed commit to fix what I mentioned above.

**Comment by ngzhian** [comment]
Oh sweet, sorry I wasn't paying attention here, thanks for fixing this!

**Comment by ngzhian** [comment]
I misread your comment (and incorrectly closed this) :) so this PR looks good to go?

**Comment by yurydelendik** [comment]
> so this PR looks good to go?

yes, I think so.

**Comment by yurydelendik** [comment]
trivial

**Pull request merged**